driver/src/main/com/mongodb/DB.java

```diff
@@ -617,4 +619,73 @@ private BsonDocument wrap(final DBObject document, final DBEncoder encoder) {
             return new BsonDocumentWrapper<DBObject>(document, new DBEncoderAdapter(encoder));
         }
     }
+
+    /**
+     * Determines the read preference that should be used for the given command.
+     *
+     * @param command             the {@link DBObject} representing the command
+     * @param requestedPreference the preference requested by the client.
+     * @return the read preference to use for the given command.  It will never return {@code null}.
+     * @see com.mongodb.ReadPreference
+     */
+    ReadPreference getCommandReadPreference(final DBObject command, final ReadPreference requestedPreference) {
+        if (mongo.getReplicaSetStatus() == null) {
```

Comment:
Is this conditional necessary anymore?  I think command operation will take care of this.